src/javaharness/java/arcs/api/NativeParticleImpl.java

```diff
@@ -1,15 +1,13 @@
 package arcs.api;
 
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 public class NativeParticleImpl implements NativeParticle {
   public ParticleSpec spec;
-  public Map<String, Handle> handles;
-
-  public NativeParticleImpl() {
-    this.handles = new HashMap();
-  }
+  public Map<String, Handle> handles = new HashMap<>();
```

Comment:
Perhaps `handlesByName` here following what you did in the next line?

Reply:
renamed to `handleByName` and `slotProxyByName` (originally i just copied the names in the ts code, but it is probably worth to have better names for data members, instead of consistency).